### PR TITLE
Auto3DSeg ensemble tests images depend on #GPUs + docstring improvement

### LINKED ISSUE #6456: Auto3DSeg tests improvement on mGPU ensemble
**Describe the bug**

It runs on single GPU, but fails on 4-GPU.

```
2023-04-28 13:02:27,188 - INFO - Ensembling using 4 GPU!
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
2023-04-28 13:02:33,112 - INFO - The output_dir is not specified. /opt/toolkit/tutorials/monai/auto3dseg/notebooks/work_dir/ensemble_output will be used to save ensemble predictions.
2023-04-28 13:02:33,126 - INFO - The output_dir is not specified. /opt/toolkit/tutorials/monai/auto3dseg/notebooks/work_dir/ensemble_output will be used to save ensemble predictions.
2023-04-28 13:02:33,159 - INFO - The output_dir is not specified. /opt/toolkit/tutorials/monai/auto3dseg/notebooks/work_dir/ensemble_output will be used to save ensemble predictions.
2023-04-28 13:02:33,272 - INFO - The output_dir is not specified. /opt/toolkit/tutorials/monai/auto3dseg/notebooks/work_dir/ensemble_output will be used to save ensemble predictions.
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/__main__.py", line 24, in <module>
    fire.Fire(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 544, in ensemble
    self.ensembler(pred_param=self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 155, in __call__
    files = files[slices]
TypeError: list indices must be integers or slices, not str
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/__main__.py", line 24, in <module>
    fire.Fire(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 544, in ensemble
    self.ensembler(pred_param=self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 155, in __call__
    files = files[slices]
TypeError: list indices must be integers or slices, not str
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/__main__.py", line 24, in <module>
    fire.Fire(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 544, in ensemble
    self.ensembler(pred_param=self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 155, in __call__
    files = files[slices]
TypeError: list indices must be integers or slices, not str
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/__main__.py", line 24, in <module>
    fire.Fire(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/usr/local/lib/python3.8/dist-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 544, in ensemble
    self.ensembler(pred_param=self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/monai/apps/auto3dseg/ensemble_builder.py", line 155, in __call__
    files = files[slices]
TypeError: list indices must be integers or slices, not str
```

The cause may be that there are only 2 testing images, and the number of testing images is less than number of GPUs

### PR DESCRIPTION
**Pull request by mingxin-zheng**
Fixes #6456.

### Description

- Make the test images also depend on GPUs
- Improve docstring for infer

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by mingxin-zheng** [comment]
Hi @wyli , I have a naive question: 

There was some docstring to show to use the `__call__` method since MONAI 1.0 in https://github.com/Project-MONAI/MONAI/blob/01552fdff6c5e95d44d2b885322f304e31000e69/monai/apps/auto3dseg/ensemble_builder.py#L132

But I can't find those in https://docs.monai.io/en/stable/apps.html#module-monai.apps.auto3dseg. Do you know why it is missing? Thanks!

**Comment by mingxin-zheng** [comment]
Thanks @wyli . I will try to set them visible then

**Review by wyli — APPROVED** [review]

File: monai/apps/auto3dseg/ensemble_builder.py [hunk 1]
```diff
@@ -136,14 +136,18 @@ def __call__(self, pred_param: dict | None = None) -> list:
...
                 parameters of the class functions.
                 The first group contains:
-                'files_slices': a value type of `slice`. The files_slices will slice the infer_files and only
-                    make prediction on the infer_files[file_slices].
-                'mode': ensemble mode. Currently "mean" and "vote" (majority voting) schemes are supported.
-                'sigmoid': use the sigmoid function (e.g. x>0.5) to convert the prediction probability map to
-                    the label class prediction, otherwise argmax(x) is used.
+                - ``"infer_files"``: file paths to the images to read in a list.
+                - ``"files_slices"``: a value type of `slice`. The files_slices will slice the ``"infer_files"`` and
+                only make prediction on the infer_files[file_slices].
```
[the remark this answers is not in the document]
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/ensemble_builder.py]: Thanks! The example is very helpful. Made edits in 65c4cf0

[the remark this answers is not in the document]
Comment (wyli) [reply monai/apps/auto3dseg/ensemble_builder.py]: I forgot that this is a nested list,  need the blank lines to separate the inner list, 
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

I'll change and verify...

**Pull request merged**